### ISSUE TITLE
Rethinking file extraction on VSIX install

### ISSUE THREAD

**Issue by crutkas** [Microsoft/WindowsTemplateStudio#2217]
Currently we place the template file and then extract and move files client side.  This adds weight and 

- Why do the files not live in the same folder as the extension?
- Can we ship the VSIX pre-expanded.
   - file size, it is a zip anyways so we still have file savings.
- Is it possible to break apart the templates into smaller bits?
   - I can't see why it wouldn't be ok for duplicates if we have a single master that we create the files from

**Comment by sibille** [comment]
We cannot add the templates without zipping them to the vsix, because all content in the vsix should be signed, so we'll have to maintain the extraction part. 
Regarding the copy: 
I tried to extract the Templates.mstx to the vsix installation folder, but got exceptions because file names exceed limits. We could shorten folder names (especially in composition templates some are very long) but we came also up with the possibility of extracting templates from the installation directory directly to the working folder (program data). 
This would allow us to skip the copy step and maintain the current template folder structure at a time.

**Comment by mrlacey** [comment]
> Extraction from vsix installation folder directly to the programData folder with splitted templates takes 10 sec for C# and 7 for VB. In comparison currently it takes 19 sec to extract and 10 to copy. I think this is the way we should go.

Just want to check there's no possible negative side-effect if this fails part way through. Is the recovery from this still ok?

**Comment by sibille** [comment]
Yep. Extraction is done to {versionnumber}_tmp directory, and then renamed, if you cancel (or sth fails) the next time you start the wizrad, the _tmp directory is deleted and the process starts over again.
